src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/ISimdVector_2.cs

```diff
@@ -154,7 +154,11 @@ internal unsafe interface ISimdVector<TSelf, T>
         /// <exception cref="ArgumentException">The length of <paramref name="destination" /> is less than <see cref="Count" />.</exception>
         /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
         /// <exception cref="NullReferenceException"><paramref name="destination" /> is <c>null</c>.</exception>
-        static virtual void CopyTo(TSelf vector, T[] destination) => TSelf.CopyTo(vector, destination.AsSpan());
+        static virtual void CopyTo(TSelf vector, T[] destination)
+        {
+            _ = destination.Length;
+            TSelf.CopyTo(vector, destination.AsSpan());
```

Comment:
`CopyTo` does perform a nullcheck, I don't see why an implicit is needed

Reply:
ah, I guess `null.AsSpan` is converted into an empty span and then it fails with ArgumentException instead of NRE?

Reply:
Then, instead of such hacks, you can do something like `new Span<T>(destination, 0, destination.Length)`

Reply:
> Then, instead of such hacks, you can do something like `new Span<T>(destination, 0, destination.Length)`

That's an implicit null check, just less obvious.

Reply:
> > Then, instead of such hacks, you can do something like `new Span<T>(destination, 0, destination.Length)`
> 
> That's also an implicit null check, just less obvious.

I think it raises less questions/reads better than a standalone discard
```
_ = destination.Length;
```

Reply:
Also, please add tests to cover this behavior

Reply:
I suggest keeping the nullcheck and adding a comment:
`// This will throw NullReferenceException if destination is null.`

Reply:
We should just do the simpler and more intuitive thing, using the span constructor does that and simplifies the overall logic since we need to create a span anyways

> (you don't have to use ISimdVector for tests, concrete vector types are enough)

We already have tests for the concrete types covering this. The issue here should only be for the internal `ISimdVector` interface which was mostly slapped together for internal use/prototyping.

Reply:
> We already have tests for the concrete types covering this. The issue here should only be for the internal `ISimdVector` interface which was mostly slapped together for internal use/prototyping.

ah, right, nvm then